File: packages/dll/src/net/messages/getstructlist.cpp
#include "getstructlist.h"
#include "../message.h"
#include "../../injector.h"
#include "../../UE/v425/uobjectarray.h"

GetStructListRequest::GetStructListRequest(Buffer& data) : Request(data) {}

void GetStructListRequest::Process() {
	GetStructListResponse res;

	res.names = injector.objectArray.GetNames();

	ProcessResponse(res);
}

void GetStructListResponse::Process(Buffer& data) {
	Response::Process(data);

	data.Write(names.size());
	for (auto& name : names)
	{
		data.Write(name);
	}
}

File: packages/electron/src/main/net/messages/getStructList.ts
import { RROxApp } from "../../app";
import { BufferIO } from "../io";
import { MessageType, Request, Response } from "../message";

export class GetStructListRequest extends Request {
    constructor( app: RROxApp ) {
        super( app, MessageType.GET_STRUCT_LIST );
    }
}

export class GetStructListResponse extends Response {
    public list: string[]

    constructor( app: RROxApp, data: BufferIO ) {
        super( app, data );

        this.list = [];
        
        data.readArray( () => {
            const str = data.readString();
            if( str )
                this.list.push( str );
        } );
    }
}